src/Platform/Microsoft.Testing.Platform/TestHostOrcherstrator/TestHostOrchestratorManager.cs

```diff
@@ -64,9 +64,9 @@ internal async Task<ITestHostOrchestratorApplicationLifetime[]> BuildTestHostOrc
             ITestHostOrchestratorApplicationLifetime service = testHostOrchestratorApplicationLifetimeFactory(serviceProvider);
 
             // Check if we have already extensions of the same type with same id registered
-            if (lifetimes.Any(x => x.Uid == service.Uid))
+            ITestHostOrchestratorApplicationLifetime? currentRegisteredExtension = lifetimes.Where(x => x.Uid == service.Uid).FirstOrDefault();
```

Comment:
We also want to throw a useful localized message if there are more than one results.